index.js

```diff
@@ -12,6 +14,19 @@ var diagnostics = require("./lib/diagnostics");
 var logs = require("./lib/logs");
 var extensions = require("./lib/extensions");
 var webjobs = require("./lib/webjobs");
+var Promise = require("bluebird");
+
+function promisifyApi(target) {
+    Object.keys(target).forEach(function (key) {
+        Promise.promisifyAll(target[key]);
+    });
+
+    Object.keys(target.extensions).forEach(function (key) {
+        Promise.promisifyAll(target.extensions[key]);
+    });
```

Comment:
I resisted the urge to do deep promisification with recursion. It just seems a bit brute force and may burn us later. For now the extensions module is the only one with nested objects.